### ISSUE TITLE
Spearbit Audit: Address contracts issues

### ISSUE THREAD

**Issue by jakekidd** [connext/nxtp#1556]
### How do
Go to:
https://github.com/spearbit-audits/connext/issues

**Go down the priority list:**
Critical>High Risk>Medium Risk>Low Risk>Gas Optimizations>Informational

**If something is related to Nomad contracts:**
- Add the "Nomad Contracts" label and "Status: Acknowledged" label
- Move on, you can ignore this for now (James will pick it up)

**If something is related to SponsorVault:**
- Add the "SponsorVault" label
- Move on, we will reassess and work on those issues once the core path issues have been addressed

### Doing a fix
- Make a PR to 1464 branch with fixes for the issue: https://github.com/connext/nxtp/pull/1466
- Once it's r4r: add the label "Fix Ready for Review" and "Status: Fixed"
- Comment it in this issue
- Feel free to tag auditors and give them a chance to review your PR if you're uncertain the fix is correct
- Once the PR is merged, remove the "Fix Ready for Review" label

### Progress
*Note: This excludes Nomad contracts and SponsorVault issues!*
- [ ] Critical
- [x] High Risk
- [ ] Medium Risk
- [ ] Low Risk
- [ ] Gas Optimizations
- [ ] Informational

**Comment by jakekidd** [comment]
r4r https://github.com/connext/nxtp/pull/1536

**Comment by jakekidd** [comment]
Discuss:

- Portal repayment swap from mad assets <> adopted assets uses oracle from our stableswap AMM pool which can be manipulated: https://github.com/spearbit-audits/connext/issues/44

- Executor can be freely looted of any stuck tokens: https://github.com/spearbit-audits/connext/issues/24

**Comment by jakekidd** [comment]
Before issue is closed out, would like to make sure the following is addressed in documentation:

> The two big conclusions from the audit in terms of policy/security that xapp developers need to know are:
> - If you have calldata that manages funds on the destination chain, xapps need to be 100% sure that there is no possibility of funds (any funds, including the tokens that were bridged and any/all other tokens) getting accidentally left in the Executor contract. Otherwise, those funds are forfeit (they can technically be scooped up by just about anyone with very little effort).
>
> - Given this new policy change: xapp developers using the callback pattern must be certain that it does not revert. Otherwise, the relayer will be paid and the callback will have been executed/completed (afaik, there aren't any "redos" here, lol).
>
> For both items above ^ we can (and should) be demonstrating an example for each of a "bad pattern" and a safe pattern

**Comment by LayneHaber** [comment]
are we okay with closing this? remediation work is done, documentation likely needs to be updated still. should that be a separate issue?

**Comment by alexwhte** [comment]
yes closing

if we need heavy documentation work please file to backlog?
